File: payroll-api/src/main/java/com/company/payroll/mapper/ResignationMapper.java
package com.company.payroll.mapper;

import java.util.List;

import org.springframework.stereotype.Repository;

import com.company.payroll.model.Resignation;

@Repository
public interface ResignationMapper {
    int deleteByPrimaryKey(Integer resignId);

    int insert(Resignation row);

    int insertSelective(Resignation row);
    
    List<Resignation> selectList();

    Resignation selectByPrimaryKey(Integer resignId);

    int updateByPrimaryKeySelective(Resignation row);

    int updateByPrimaryKey(Resignation row);
}

File: kotlin-payroll/src/main/kotlin/com/company/payroll/mapper/ResignationMapper.kt
package com.company.payroll.mapper

import com.company.payroll.model.Resignation
import org.springframework.stereotype.Repository

@Repository
interface ResignationMapper {
  fun deleteByPrimaryKey(resignId: Int): Int
  fun insert(row: Resignation): Int
  fun insertSelective(row: Resignation): Int
  fun selectList(): List<Resignation>?
  fun selectByPrimaryKey(resignId: Int): Resignation?
  fun updateByPrimaryKeySelective(row: Resignation): Int
  fun updateByPrimaryKey(row: Resignation): Int
}